pdr_backend/lake/csv_data_store.py

```diff
@@ -125,6 +126,11 @@ def write(
 
```

Comment:
(This is a general comment for csv_data_store.py)

75% of the methods inside `CSVDataStore` don't update the state of a `CSVDataStore` object. And most of those don't need to access any of the class's data (`self.x`) either. By having the methods inside `CSVDataStore`, it makes the class thicker than it needs to be, and therefore harder to test and maintain.

Better: change each method to a function, in a new module `cvsutil.py`. Each function should do exactly one thing well, be easy to understand, and be well-tested (and therefore hardened). Just like we have traditionally done with plutil.py. Below I reference previous examples that are actually csv utilities, you can use those functions (and their testing code) too.

The result:
- the `CSVDataStore` class will be far smaller and easy to understand
- each utility function will be small, easy to understand, and well-tested

Reply:
I apologize for any confusion earlier. 

The CSVDataStore is not developed to provide basic helper functions for reading data from a CSV file. 

Rather, it is a structured CSV data store designed to be used by the application. By utilizing this class, the application layer can avoid dealing with CSV-related tasks, as everything will be handled by the CSVDataStore, including partitioning, foldering, storing and accessing data (for specific periods too).

All the application layer needs to do is provide the dataset name, and using related methods it will receive (or store) the desired data.